### ISSUE TITLE
Tab icons not rendering properly after 2.11 update, suspected due to breaking changes in data.json

### ISSUE THREAD

**Issue by rx-chris** [FlorianWoelki/obsidian-iconize#404]
**Describe the bug**
Tab icons not rendering properly and exhibits the following behavior:
1. Tab icon renders right after '`icon`' frontmatter value is changed but...
2. Tab icon stop rendering after leaving the page and coming back again or opening it in a new tab.
3. If I inspect the Tab icon element in the developer console, the element seems to be correct but it just does not render for some reason.
4. There are also cases where the inspected Tab icon was using the icon of the previous page after opening a new page in the tab (but still does not render properly)
5. This behaviour only seems to affect notes that have their icons changed in the frontmatter after the 2.11 update

**To Reproduce**
After doing a few rounds of testing I suspect the problem is in the `data.json` file, more specifically with the new way of storing frontmatter icon data:
- Old method (which seems to work better without much issue):
  ```js
  "Notes/Tag Page.md": "TiHash"
  ```
- New method (suspect to be cause of tab rendering issues):
  ```js
  "Notes/Tag Page.md": {
      "iconName": "TiHash"
  }
  ```

**Expected behaviour**
- Tab icons should render fine even after opening a new tab

**Temporary Solution**
- clean up the `data.json` to the old method and then revert back to 2.10 version

**Screenshots**
![image](https://github.com/FlorianWoelki/obsidian-iconize/assets/146159597/f20406fd-b61b-4f57-8286-ecb107eb8f3c)

**Comment by rx-chris** [comment]
An alternative solution that I tried and seemingly worked with 2.11 is to open the `data.json` file in Visual Studio Code and do a regex search and replace to convert the old icon format to the new Icon format.

- regex search for old format that I used: `"(?:Si|Li|Fab|Ti|Far|Fas).+"`
- replace to new format: `{"iconName": $0}`

Although it works, it is not the most elegant solution since I had to manually modify the `data.json` file. Furthermore, if I were to delete the `data.json` file, the frontmatter icon data is no longer remembered by Iconize even if I relaunch Obsidian. I have raised this as a separate issue in #423 .
